src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheckTest.java

```diff
@@ -21,14 +21,22 @@
 
 import static com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocPackageCheck.MSG_LEGACY_PACKAGE_HTML;
 import static com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocPackageCheck.MSG_PACKAGE_INFO;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
 
 import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
 
 import org.junit.Test;
 
 import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
 import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
 import com.puppycrawl.tools.checkstyle.api.Configuration;
+import com.puppycrawl.tools.checkstyle.api.FileText;
 import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
 
 public class JavadocPackageCheckTest
```

Comment:
Can we reproduce exception from the issue?

Reply:
final File fileWithInvalidName = new File("\u0000\u0000\u0000");
        final FileText dummyFileText = new FileText(fileWithInvalidName, Collections.emptyList());
        try {
            check.processFiltered(fileWithInvalidName, dummyFileText);

Reproduced exception using real File object.

Reply:
This issue is CLI-dependent. If we want to test how do checks behave with different path patterns we probably should create common test method in AbstractModuleTestSupport. Which is out of scope of current task.

Reply:
@AADudkin ,
as problem is reproducible by CLI - there should be UT case, exactly as use provided and at least at MainTest , but ideally all should be at /JavadocPackageCheckTest.java .

Reply:
Added test case to JavadocPackageCheckTest.

Reply:
ok